code/week5/week5.cpp

```diff
@@ -0,0 +1,106 @@
+/*
+ * to compile
+ * g++ -std=c++14 -o week5.out *.cpp
+ */
+#include <iostream>
```

Comment:
some compilers auto-include "useful" libraries and print a warning, for example, gcc will auto-include iostream with a warning, so that might be it. :P